ml-agents/mlagents/trainers/torch/networks.py

```diff
@@ -137,22 +139,20 @@ def forward(
         memories: Optional[torch.Tensor] = None,
         sequence_length: int = 1,
     ) -> Tuple[Dict[str, torch.Tensor], torch.Tensor]:
-        embedding, memories = self.network_body(
+        encoding, memories = self.network_body(
             vec_inputs, vis_inputs, actions, memories, sequence_length
         )
-        output, _ = self.value_heads(embedding)
+        output = self.value_heads(encoding)
         return output, memories
 
 
-class ActorCritic(nn.Module):
+class Actor(nn.Module):
```

Comment:
I think Actor should be an interface (not inherit from nn.Module and only have abstract methods). Then have a "SimpleActor" if we need a base implementation. I think that would make the API on the Actor much clearer

Reply:
Would you still do `Actor -> SimpleActor -> ActorCritic -> SeparateActorCritic`?

Reply:
Made some changes - it's now `SimpleActor(nn.Module, Actor)` and the other two inherit from that.